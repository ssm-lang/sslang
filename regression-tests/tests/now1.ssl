us t = 1000 * t
now_us _ = now / 1000

logger n period current_time start_time kill =
  let n_ = new n
  let event = new ()
  while deref n_ > 0
    current_time <- now_us () - start_time
    after period, event <- ()
    wait event
    n_ <- deref n_ - 1
  kill <- deref kill - 1

putip putc x =
  if x < 10
    putc (x + 48)
  else
    putip putc (x / 10)
    putc (48 + x % 10)

main cin cout =
  let putc c =
       after 1, cout <- c
       wait cout
  let kill = new 2
  let display r =
       wait r
            kill
       while deref kill > 0
         putip putc (deref r)
         putc 10
         wait r
              kill
  let current_time = new (now_us ())
  par logger 30 (us 50) current_time (deref current_time) kill
      logger 50 (us 30) current_time (deref current_time) kill
      display current_time
