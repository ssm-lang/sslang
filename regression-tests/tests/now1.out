30
50
60
90
100
120
150
180
200
210
240
250
270
300
330
350
360
390
400
420
450
480
500
510
540
550
570
600
630
650
660
690
700
720
750
780
800
810
840
850
870
900
930
950
960
990
1000
1020
1050
1080
1100
1110
1140
1150
1170
1200
1230
1250
1260
1290
1300
1320
1350
1380
1400
1410
1440
1450
1470
